src/hotspot/share/prims/jni.cpp

```diff
@@ -444,9 +444,10 @@ JNI_ENTRY(jobject, jni_ToReflectedMethod(JNIEnv *env, jclass cls, jmethodID meth
   methodHandle m (THREAD, Method::resolve_jmethod_id(method_id));
   assert(m->is_static() == (isStatic != 0), "jni_ToReflectedMethod access flags doesn't match");
   oop reflection_method;
-  if (m->is_initializer()) {
+  if (m->is_object_initializer()) {
     reflection_method = Reflection::new_constructor(m, CHECK_NULL);
   } else {
+    assert(!m->is_static_initializer(), "Cannot be static initializer");
```

Comment:
This looks like a behavioral change; otherwise reflection and method handle changes look good. Per JNI https://docs.oracle.com/en/java/javase/23/docs/specs/jni/functions.html#toreflectedmethod it seems to just return if the method id is valid.

Reply:
You mean the _assert_ is excessive, or something else? I think it is within the spec to return method for `clinit`-s, since clinits are logically not constructors.

Reply:
Then this assert seems excessive. If JNI users get the method slot (presumably through some hacks) for a `<clinit>` then JNI has to return the `<clinit>` and this assert doesn't stand.

Reply:
OK, I replaced the assert with a comment.